src/manifest.c

```diff
@@ -198,6 +201,14 @@ static gboolean parse_manifest(GKeyFile *key_file, RaucManifest **manifest, GErr
 	tmp = key_file_consume_string(key_file, "bundle", "format", NULL);
 	if (tmp == NULL || g_strcmp0(tmp, "plain") == 0) {
 		raucm->bundle_format = R_MANIFEST_FORMAT_PLAIN;
+		g_ptr_array_add(raucm->warnings, g_strdup(
```

Comment:
Two issues here:
* The first is that when being within this if statement, the warning will still be printed even if 'plain' was selected explicitly. This is not what the warning message suggests.

  This should be solvable by handling `tmp == NULL` and `g_strcmp0(tmp, "plain") == 0)` in different if statements and emitting the warning only on `tmp == NULL`.
* The second is that the manifest parsing is not idempotent on a `format=plain` since when writing the manifest back, a `format=plain` will just be ignored and the `[bundle]` section will thus be deleted. This means even if we had set `format=plain` explicitly, a second call would re-trigger the warning.

  If the parsing would always write `format=plain` then we would have the same issue but the other way around.